.github/workflows/dependency-upgrade.yaml

```diff
@@ -0,0 +1,72 @@
+---
+name: Dependency Upgrade Check
+
+on:
+  schedule:
+    # Runs daily at noon UTC (I guess)
+    - cron:  '0 12 * * *'
+
+jobs:
+  check-upgrades:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        dependency: ["Stargate", "Medusa", "Reaper"]
+        include:
+          - dependency: Stargate
+            version_path: ".stargate.version"
+            docker_hub_url: "https://registry.hub.docker.com/v2/repositories/stargateio/stargate-3_11/tags?page_size=1"
+          - dependency: Medusa
+            version_path: ".medusa.image.tag"
+            docker_hub_url: "https://registry.hub.docker.com/v2/repositories/k8ssandra/medusa/tags?page_size=25"
+          - dependency: Reaper
+            version_path: ".reaper.image.tag"
+            docker_hub_url: "https://registry.hub.docker.com/v2/repositories/thelastpickle/cassandra-reaper/tags?page_size=25"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Update PATH
+        run: |
+          echo "$GITHUB_WORKSPACE/bin" >> $GITHUB_PATH
+      - name: Install YQ and JQ
+        run: |
+          scripts/install-yq.sh v4.6.3
+          yq -V
+          sudo apt-get update
+          sudo apt-get install jq -y
+          jq --version
+      - name: Get current dependencies versions
+        run: |
+          CURRENT=$(yq e '${{ matrix.version_path }}' charts/k8ssandra/values.yaml)
+          
+          echo "CURRENT_VERSION=${CURRENT}" >> $GITHUB_ENV
+          
+      - name: Get latest released versions
+        run: |
+          LATEST=$(curl -L -s '${{ matrix.docker_hub_url }}'|jq -r '.results[]["name"]'|grep -Ei "^[v]{0,1}[0-9]{1,2}\.[0-9]{1,2}\.[0-9]{1,2}$"|head -1|sed s/v//)
+
+          echo "LATEST_VERSION=${LATEST}" >> $GITHUB_ENV
+          
+      - name: Check if an open PR exists for the latest versions
```

Comment:
Won't the create-pull-request action do this? At least it says:

``If a pull request already exists and there are no further changes (i.e. no diff with the current pull request branch) then the action exits silently.``

Reply:
I've seen that, yes, but I was afraid it may reopen a PR that we close if we don't want to perform the upgrade 🤔 
Let me try this actually and we'll see if it doesn't try to reopen closed PRs.